### ISSUE TITLE
Improvement of the note editing dialog

### ISSUE THREAD

**Issue by swiftech** [raydac/netbeans-mmd-plugin#86]
Once upon a  time, I opened a note dialog and typed something into the content, then I leaved this dialog open to something else, when I came back, I forgot what I did before and accidentally  pressed the 'ESC' button, everything that I added to the content was lost. So could you add a new 'Save' button to the dialog and check the changes of content before closing it?

version: sciareto 1.5.0

**Comment by raydac** [comment]
I have made improvements, you can try snapshot

**Comment by swiftech** [comment]
the new version can not be opened on macOS, probably something wrong with the packaging process.

<img width="419" alt="Screen Shot 2021-08-10 at 10 23 40 AM" src="https://user-images.githubusercontent.com/3277107/128798929-ee459c19-f397-43f3-8eb0-173e6ec39641.png">
<img width="365" alt="Screen Shot 2021-08-10 at 10 24 24 AM" src="https://user-images.githubusercontent.com/3277107/128798967-df39ab0d-a0b0-4def-b562-b88d04109c1c.png">

**Comment by swiftech** [comment]
yes, the checking for 'esc' button works now, but I think it should also works on the "cancel" button of the dialog.

besides, If the "saving to disk" can be performed during the dialog opening, it will be very useful for me because sometimes I use the note dialog to edit very long content. 

thanks.

**Comment by raydac** [comment]
I have added functionality for cancel button and window close button in SciaReto, you can try snapshot
